web/src/compose_ui.ts

```diff
@@ -383,13 +385,16 @@ export function handle_keydown(
     // key was on. We turn to key to lowercase so the key bindings
     // work regardless of whether Caps Lock was on or not.
     const key = event.key.toLowerCase();
+    const is_empty_selection = $textarea.range().text.length === 0;
```

Comment:
Let's try to avoid doing this calculation in the event that a relevant key wasn't pressed.

(Typing in the compose area is a code path where micro-optimizations like this are worth it, since lag is very annoying)